<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TabFormat</class>
 <widget class="QWidget" name="TabFormat">
  <property name="geometry">
   <rect><x>0</x><y>0</y><width>1000</width><height>700</height></rect>
  </property>
  <layout class="QHBoxLayout" name="mainLayout">
   <property name="spacing"><number>8</number></property>
   <property name="leftMargin"><number>4</number></property>
   <property name="topMargin"><number>4</number></property>
   <property name="rightMargin"><number>4</number></property>
   <property name="bottomMargin"><number>4</number></property>
   
   <!-- ═══════════════════ COLUMN 1 (33%) ═══════════════════ -->
   <item>
    <widget class="QWidget" name="col1Widget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QVBoxLayout" name="col1">
      <property name="spacing"><number>4</number></property>
      <property name="leftMargin"><number>0</number></property>
      <property name="topMargin"><number>0</number></property>
      <property name="rightMargin"><number>0</number></property>
      <property name="bottomMargin"><number>0</number></property>
      
      <!-- SYSTEM / FORMAT -->
      <item>
       <widget class="QGroupBox" name="groupSystem">
        <property name="minimumSize"><size><width>0</width><height>100</height></size></property>
        <property name="title"><string>System / Format</string></property>
        <layout class="QFormLayout">
         <item row="0" column="0"><widget class="QLabel"><property name="text"><string>System:</string></property></widget></item>
         <item row="0" column="1"><widget class="QComboBox" name="comboSystem"/></item>
         <item row="1" column="0"><widget class="QLabel"><property name="text"><string>Format:</string></property></widget></item>
         <item row="1" column="1"><widget class="QComboBox" name="comboFormat"/></item>
         <item row="2" column="0"><widget class="QLabel"><property name="text"><string>Version:</string></property></widget></item>
         <item row="2" column="1"><widget class="QComboBox" name="comboVersion"/></item>
        </layout>
       </widget>
      </item>
      
      <!-- FORMAT -->
      <item>
       <widget class="QGroupBox" name="groupFormat">
        <property name="minimumSize"><size><width>0</width><height>100</height></size></property>
        <property name="title"><string>Format</string></property>
        <layout class="QGridLayout">
         <item row="0" column="0"><widget class="QLabel"><property name="text"><string>Tracks:</string></property></widget></item>
         <item row="0" column="1"><widget class="QSpinBox" name="spinTracks"><property name="maximum"><number>255</number></property><property name="value"><number>80</number></property></widget></item>
         <item row="0" column="2"><widget class="QLabel"><property name="text"><string>Sides:</string></property></widget></item>
         <item row="0" column="3"><widget class="QSpinBox" name="spinSides"><property name="minimum"><number>1</number></property><property name="maximum"><number>2</number></property><property name="value"><number>1</number></property></widget></item>
         <item row="1" column="0"><widget class="QLabel"><property name="text"><string>Sectors:</string></property></widget></item>
         <item row="1" column="1"><widget class="QSpinBox" name="spinSectors"><property name="minimum"><number>1</number></property><property name="maximum"><number>36</number></property><property name="value"><number>11</number></property></widget></item>
         <item row="1" column="2"><widget class="QLabel"><property name="text"><string>Size:</string></property></widget></item>
         <item row="1" column="3"><widget class="QComboBox" name="comboSectorSize">
          <item><property name="text"><string>256</string></property></item>
          <item><property name="text"><string>512</string></property></item>
         </widget></item>
         <item row="2" column="0"><widget class="QLabel"><property name="text"><string>Encoding:</string></property></widget></item>
         <item row="2" column="1"><widget class="QComboBox" name="comboEncoding">
          <item><property name="text"><string>MFM</string></property></item>
          <item><property name="text"><string>FM</string></property></item>
          <item><property name="text"><string>GCR</string></property></item>
         </widget></item>
         <item row="2" column="2"><widget class="QLabel"><property name="text"><string>RPM:</string></property></widget></item>
         <item row="2" column="3"><widget class="QComboBox" name="comboRPM">
          <item><property name="text"><string>300</string></property></item>
          <item><property name="text"><string>360</string></property></item>
         </widget></item>
        </layout>
       </widget>
      </item>
      
      <!-- XCOPY -->
      <item>
       <widget class="QGroupBox" name="groupXCopy">
        <property name="minimumSize"><size><width>0</width><height>100</height></size></property>
        <property name="title"><string>XCopy</string></property>
        <layout class="QVBoxLayout">
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLabel"><property name="text"><string>Start:</string></property></widget></item>
           <item><widget class="QSpinBox" name="spinStartTrack"><property name="maximum"><number>255</number></property></widget></item>
           <item><widget class="QLabel"><property name="text"><string>End:</string></property></widget></item>
           <item><widget class="QSpinBox" name="spinEndTrack"><property name="maximum"><number>255</number></property><property name="value"><number>79</number></property></widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLabel"><property name="text"><string>Sides:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboXCopySides">
            <item><property name="text"><string>Both</string></property></item>
            <item><property name="text"><string>Side 0</string></property></item>
            <item><property name="text"><string>Side 1</string></property></item>
           </widget></item>
           <item><widget class="QLabel"><property name="text"><string>Mode:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboXCopyMode">
            <item><property name="text"><string>Sector</string></property></item>
            <item><property name="text"><string>Track</string></property></item>
            <item><property name="text"><string>Index</string></property></item>
           </widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QCheckBox" name="checkXCopyVerify"><property name="text"><string>Verify</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item><widget class="QCheckBox" name="checkAllTracks"><property name="text"><string>All tracks</string></property><property name="checked"><bool>true</bool></property></widget></item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      
      <!-- PRESETS -->
      <item>
       <widget class="QGroupBox" name="groupPresets">
        <property name="minimumSize"><size><width>0</width><height>80</height></size></property>
        <property name="title"><string>Presets</string></property>
        <layout class="QVBoxLayout">
         <item><widget class="QComboBox" name="comboPreset"/></item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QPushButton" name="btnLoadPreset"><property name="text"><string>Load</string></property></widget></item>
           <item><widget class="QPushButton" name="btnSavePreset"><property name="text"><string>Save</string></property></widget></item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      
      <item><spacer name="col1Spacer"><property name="orientation"><enum>Qt::Vertical</enum></property><property name="sizeHint" stdset="0"><size><width>20</width><height>40</height></size></property></spacer></item>
     </layout>
    </widget>
   </item>
   
   <!-- ═══════════════════ COLUMN 2 (33%) ═══════════════════ -->
   <item>
    <widget class="QWidget" name="col2Widget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QVBoxLayout" name="col2">
      <property name="spacing"><number>4</number></property>
      <property name="leftMargin"><number>0</number></property>
      <property name="topMargin"><number>0</number></property>
      <property name="rightMargin"><number>0</number></property>
      <property name="bottomMargin"><number>0</number></property>
      
      <!-- FLUX -->
      <item>
       <widget class="QGroupBox" name="groupFlux">
        <property name="minimumSize"><size><width>0</width><height>100</height></size></property>
        <property name="title"><string>Flux</string></property>
        <layout class="QVBoxLayout">
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLabel"><property name="text"><string>Speed:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboFluxSpeed">
            <item><property name="text"><string>Safe</string></property></item>
            <item><property name="text"><string>Normal</string></property></item>
            <item><property name="text"><string>Fast</string></property></item>
           </widget></item>
           <item><widget class="QLabel"><property name="text"><string>Revs:</string></property></widget></item>
           <item><widget class="QSpinBox" name="spinRevolutions"><property name="minimum"><number>1</number></property><property name="maximum"><number>10</number></property><property name="value"><number>3</number></property></widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLabel"><property name="text"><string>Errors:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboFluxErrors">
            <item><property name="text"><string>Strict</string></property></item>
            <item><property name="text"><string>Relaxed</string></property></item>
            <item><property name="text"><string>Ignore</string></property></item>
           </widget></item>
           <item><widget class="QLabel"><property name="text"><string>Merge:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboFluxMerge">
            <item><property name="text"><string>First</string></property></item>
            <item><property name="text"><string>Best</string></property></item>
            <item><property name="text"><string>All</string></property></item>
           </widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QCheckBox" name="checkWeakBits"><property name="text"><string>Weak bits</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item><widget class="QCheckBox" name="checkNoFluxAreas"><property name="text"><string>No-flux areas</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item><widget class="QPushButton" name="btnFluxAdvanced"><property name="text"><string>Advanced...</string></property></widget></item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      
      <!-- PLL -->
      <item>
       <widget class="QGroupBox" name="groupPLL">
        <property name="minimumSize"><size><width>0</width><height>80</height></size></property>
        <property name="title"><string>PLL</string></property>
        <layout class="QVBoxLayout">
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLabel"><property name="text"><string>Sample:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboSampleRate">
            <item><property name="text"><string>Auto</string></property></item>
            <item><property name="text"><string>25 MHz</string></property></item>
            <item><property name="text"><string>50 MHz</string></property></item>
            <item><property name="text"><string>80 MHz</string></property></item>
           </widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QCheckBox" name="checkAdaptivePLL"><property name="text"><string>Adaptive PLL</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item><widget class="QCheckBox" name="checkUseIndex"><property name="text"><string>Use index</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item><widget class="QPushButton" name="btnPLLAdvanced"><property name="text"><string>Advanced...</string></property></widget></item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      
      <!-- GCR / NIBBLE -->
      <item>
       <widget class="QGroupBox" name="groupNibble">
        <property name="minimumSize"><size><width>0</width><height>100</height></size></property>
        <property name="title"><string>GCR / Nibble</string></property>
        <layout class="QVBoxLayout">
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLabel"><property name="text"><string>Type:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboGCRType">
            <item><property name="text"><string>Standard GCR</string></property></item>
            <item><property name="text"><string>Apple II GCR</string></property></item>
            <item><property name="text"><string>C64 GCR</string></property></item>
           </widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QCheckBox" name="checkNibbleMode"><property name="text"><string>Nibble mode (raw)</string></property></widget></item>
           <item><widget class="QCheckBox" name="checkDecodeGCR"><property name="text"><string>Decode GCR</string></property><property name="checked"><bool>true</bool></property></widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QCheckBox" name="checkHalfTracks"><property name="text"><string>Half-tracks</string></property></widget></item>
           <item><widget class="QPushButton" name="btnNibbleAdvanced"><property name="text"><string>Advanced...</string></property></widget></item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      
      <!-- WRITE -->
      <item>
       <widget class="QGroupBox" name="groupWrite">
        <property name="minimumSize"><size><width>0</width><height>80</height></size></property>
        <property name="title"><string>Write</string></property>
        <layout class="QVBoxLayout">
         <item><widget class="QCheckBox" name="checkWritePrecomp"><property name="text"><string>Write precompensation</string></property></widget></item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLabel"><property name="text"><string>Platform:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboPlatform">
            <item><property name="text"><string>Auto-detect</string></property></item>
            <item><property name="text"><string>Commodore</string></property></item>
            <item><property name="text"><string>Amiga</string></property></item>
            <item><property name="text"><string>Atari</string></property></item>
            <item><property name="text"><string>Apple</string></property></item>
            <item><property name="text"><string>PC/DOS</string></property></item>
           </widget></item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      
      <item><spacer name="col2Spacer"><property name="orientation"><enum>Qt::Vertical</enum></property><property name="sizeHint" stdset="0"><size><width>20</width><height>40</height></size></property></spacer></item>
     </layout>
    </widget>
   </item>
   
   <!-- ═══════════════════ COLUMN 3 (33%) ═══════════════════ -->
   <item>
    <widget class="QWidget" name="col3Widget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QVBoxLayout" name="col3">
      <property name="spacing"><number>4</number></property>
      <property name="leftMargin"><number>0</number></property>
      <property name="topMargin"><number>0</number></property>
      <property name="rightMargin"><number>0</number></property>
      <property name="bottomMargin"><number>0</number></property>
      
      <!-- PROTECTION -->
      <item>
       <widget class="QGroupBox" name="groupProtection">
        <property name="minimumSize"><size><width>0</width><height>140</height></size></property>
        <property name="title"><string>Protection Detection</string></property>
        <layout class="QVBoxLayout">
         <item><widget class="QCheckBox" name="checkDetectAll"><property name="text"><string>Detect all protections</string></property><property name="checked"><bool>true</bool></property></widget></item>
         <item>
          <layout class="QGridLayout">
           <item row="0" column="0"><widget class="QCheckBox" name="checkDetectWeakBitsProt"><property name="text"><string>Weak bits</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item row="0" column="1"><widget class="QCheckBox" name="checkDetectLongTracks"><property name="text"><string>Long tracks</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item row="1" column="0"><widget class="QCheckBox" name="checkDetectHalfTracks"><property name="text"><string>Half-tracks</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item row="1" column="1"><widget class="QCheckBox" name="checkDetectTiming"><property name="text"><string>Timing var.</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item row="2" column="0"><widget class="QCheckBox" name="checkDetectNoFlux"><property name="text"><string>No-flux areas</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item row="2" column="1"><widget class="QCheckBox" name="checkDetectCustomSync"><property name="text"><string>Custom sync</string></property><property name="checked"><bool>true</bool></property></widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QRadioButton" name="radioPreserve"><property name="text"><string>Preserve</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item><widget class="QRadioButton" name="radioRemove"><property name="text"><string>Remove</string></property></widget></item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      
      <!-- FORENSIC -->
      <item>
       <widget class="QGroupBox" name="groupForensic">
        <property name="minimumSize"><size><width>0</width><height>100</height></size></property>
        <property name="title"><string>Forensic</string></property>
        <layout class="QVBoxLayout">
         <item><widget class="QCheckBox" name="checkGenerateHash"><property name="text"><string>Generate hash (SHA-256)</string></property><property name="checked"><bool>true</bool></property></widget></item>
         <item><widget class="QCheckBox" name="checkGenerateReport"><property name="text"><string>Generate report</string></property></widget></item>
         <item><widget class="QCheckBox" name="checkStrictMode"><property name="text"><string>Strict mode (no guessing)</string></property></widget></item>
        </layout>
       </widget>
      </item>
      
      <!-- LOGGING -->
      <item>
       <widget class="QGroupBox" name="groupLogging">
        <property name="minimumSize"><size><width>0</width><height>130</height></size></property>
        <property name="title"><string>Logging</string></property>
        <layout class="QVBoxLayout">
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLabel"><property name="text"><string>Level:</string></property></widget></item>
           <item><widget class="QComboBox" name="comboLogLevel">
            <item><property name="text"><string>Off</string></property></item>
            <item><property name="text"><string>Error</string></property></item>
            <item><property name="text"><string>Warning</string></property></item>
            <item><property name="text"><string>Info</string></property></item>
            <item><property name="text"><string>Debug</string></property></item>
           </widget></item>
          </layout>
         </item>
         <item><widget class="QCheckBox" name="checkLogToFile"><property name="text"><string>Log to file</string></property></widget></item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QLineEdit" name="editLogPath"><property name="placeholderText"><string>Log file path...</string></property><property name="enabled"><bool>false</bool></property></widget></item>
           <item><widget class="QPushButton" name="btnBrowseLog"><property name="text"><string>...</string></property><property name="maximumWidth"><number>30</number></property><property name="enabled"><bool>false</bool></property></widget></item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout">
           <item><widget class="QCheckBox" name="checkLogTimestamps"><property name="text"><string>Timestamps</string></property><property name="checked"><bool>true</bool></property></widget></item>
           <item><widget class="QCheckBox" name="checkVerboseLog"><property name="text"><string>Verbose</string></property></widget></item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      
      <item><spacer name="col3Spacer"><property name="orientation"><enum>Qt::Vertical</enum></property><property name="sizeHint" stdset="0"><size><width>20</width><height>40</height></size></property></spacer></item>
     </layout>
    </widget>
   </item>
   
  </layout>
 </widget>
</ui>
